src/service/mercury/index.ts

```diff
@@ -604,12 +633,30 @@ export class MercuryClient {
       if (!hasIndexerSupport(network)) {
         throw new Error(`network not currently supported: ${network}`);
       }
+
       const subs = await this.getAccountSubForPubKey(pubKey, network);
       const hasSubs = hasSubForPublicKey(subs, pubKey);
       if (!hasSubs) {
         throw new Error(ERROR.MISSING_SUB_FOR_PUBKEY);
       }
 
+      const tokenDetails = {} as {
+        [index: string]: Awaited<ReturnType<MercuryClient["tokenDetails"]>>;
+      };
+      for (const contractId of contractIds) {
+        const tokenSubs = await this.getTokenBalanceSub(
+          pubKey,
+          contractId,
+          network
+        );
+        const hasTokenSubs = hasSubForTokenBalance(tokenSubs, contractId);
```

Comment:
This check should have been added with the previous PR to add missing subscription checks. We fallback to Horizon/RPC if the account balance query includes tokens that have no subscriptions.